bundles/org.openhab.binding.proteusecometer/src/main/java/org/openhab/binding/proteusecometer/internal/ProteusEcoMeterConfiguration.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.proteusecometer.internal;
+
+/**
+ * The {@link ProteusEcoMeterConfiguration} class contains fields mapping thing configuration parameters.
+ *
+ * @author Matthias Herrmann - Initial contribution
+ */
+public class ProteusEcoMeterConfiguration {
+    public String usbPort;
```

Comment:
```suggestion
@NonNullByDefault
public class ProteusEcoMeterConfiguration {
    public String usbPort="";
```
Removes the need to check for null and NPE can not occur.

Reply:
.. in exchange for a non type universal way of representing "this is not defined", hindering parametric polymorphism and therefore the most general (and reusable) class of algorithms we know of.

But you are right, in this system it's probably the better tradeoff and I will apply it. I just come from a universe where "is not defined" is represented explicitly and in a principled type universal fashion such that problems like this one just don't occur. This is not meant as an offend :)

Reply:
No offence taken. I don’t know much on this topic as I am self taught and only worked on this project take what I suggest as a suggestion and not a request as you probably know more then me.

Reply:
@Skinah I have lots of respect for self taught people. If you are interested in other ways to write typesafe programs, take a look at https://typelevel.org/ and especially http://typelevel.org/cats/